File: app/src/main/java/com/example/task2/models/Benchmark.java
package com.example.task2.models;

import java.util.List;
import java.util.Random;

public interface Benchmark {
    String specificNumber = "28";
    Random random = new Random();

    int getNumberOfColumns();

    List<CellOperation> createItemsList(boolean setRunning);

    long measureTime(CellOperation item, int number);
}

File: app/src/main/java/com/example/task2/models/benchmarks/Benchmark.kt
package com.example.task2.models.benchmarks

import com.example.task2.models.room.ResultData

interface Benchmark {

    fun getNumberOfColumns(): Int

    fun createItemsList(setRunning: Boolean): List<CellOperation>

    fun measureTime(item: CellOperation, number: Int): Long

    fun setData(action: String, type: String, time: Long, input: Int)

    fun getData(): List<ResultData>
}
